add level to #some-component, proper $target for lazy-render-wrapper, tests

a15n opened the pull request:
this addresses the bugs identified in this ticket https://github.com/sir-dunxalot/ember-tooltips/issues/138

notale improvements...
* lazy-render-wrapper [will now get](https://github.com/sir-dunxalot/ember-tooltips/pull/149/files#diff-8096b1ffac251e79ea18b7393d9f241fR145) the correct $target element, which may be different when using `<x>-on-component`, `<x>-on-element`, and when `target="some-id"`
  * [before](https://github.com/sir-dunxalot/ember-tooltips/pull/149/files#diff-8096b1ffac251e79ea18b7393d9f241fL148) it was always getting the parentElement
* renamed `entryInteractionEvents` to `_lazyRenderEvents`
* added [_targetEventNameSpace](https://github.com/sir-dunxalot/ember-tooltips/pull/149/files#diff-8096b1ffac251e79ea18b7393d9f241fR163) to keep the code DRY
* willDestroyElement also references `$target` to clean up event handlers
* added an empty `<div>` to `some-component`, which adds extra robustness for `-on-component` tests while maintaining the same UI in the dummy site
* added `render-events-test.js`

sir-dunxalot commented on addon/components/lazy-render-wrapper.js:
```diff
@@ -120,46 +120,72 @@ export default Ember.Component.extend({
 
 	event: 'hover', // hover, click, focus, none
-	entryInteractionEvents: computed('event', function() {
+	_lazyRenderEvents: computed('event', function() {
```
I like this improvement!

a15n replied:
picking variable names is hard. I'm happy with this one though

sir-dunxalot commented on addon/components/lazy-render-wrapper.js:
```diff
@@ -120,46 +120,72 @@ export default Ember.Component.extend({
+		return _lazyRenderEvents;
+	}),
+
+	$target: computed('target', 'tetherComponentName', function() {
```
Can you add a bit of documentation to describe why this CP is necessary? I like the style of the [YUI Docs Syntax](http://yui.github.io/yuidoc/syntax/index.html) and have tried to use that in a couple of places. What do you think?

sir-dunxalot replied:
And when I say "documentation" I mean code comments documentation and not readme documentation.

a15n replied:
I think that's a good idea. I'll add some docs here in the YUI style. I think we should go through all the code and add docs to the complex properties and methods at some point.

sir-dunxalot commented on addon/components/lazy-render-wrapper.js:
```diff
@@ -120,46 +120,72 @@ export default Ember.Component.extend({
+		} else if (this.get('tetherComponentName').indexOf('-on-component') >= 0) {
+			// TODO(Andrew) refactor this once we've gotten rid of the -on-component approach
+			// share the functionality with `onComponentTarget`
+			const targetViewElementId = this.get('parentView.elementId');
```
Will there always be a `parentView` here?

a15n replied:
I just copied over the console.warn checks from the utils file. Better safe than sorry. I'm OK with copying because I'll be cleaning this up when we get rid of the `<x>-on-component` components in favor of `ember-tooltip` with a boolean.

sir-dunxalot commented on addon/components/lazy-render-wrapper.js:
```diff
@@ -120,46 +120,72 @@ export default Ember.Component.extend({
+		return $target;
 	}),
 
+	_targetEventNameSpace: 'target-lazy-render-wrapper',
```
Should this be a `const` in the file instead of a property on the component? That would make it truly private and would still avoid repetitions of the classname in the code.

sir-dunxalot commented on tests/integration/components/render-events-test.js:
```diff
@@ -0,0 +1,127 @@
+        'the eventHandler\'s namespace property be unique to ember-tooltips');
+  }
+
+  function assertNumEventsExist(num) {
```
I think the use of these helpers is self-explanatory but would like at least one line of comments to explain why and when these function should be used. This would enabled contributors to more easily understand and add to these files.

a15n commented:
Thanks for the review. I've addressed the feedback here. I think this is ready to merge.

https://github.com/sir-dunxalot/ember-tooltips/pull/149/commits/56eedd6da6c3712077dc2d0f0ef12132c2acb47e

sir-dunxalot commented:
Released as `2.6.1`.

sir-dunxalot commented:
[Release notes here](https://github.com/sir-dunxalot/ember-tooltips/releases/tag/2.6.1).
